File: lab/javascript/tecnicas/fila_enqueue_dequeue.js
const queue = [];  // Fila vazia

function enqueue(x) {
    queue.push(x);  // enfileira
}

function dequeue() {
    if (queue.length === 0) throw new Error("FILA VAZIA!");
    return queue.splice(0, 1);
}

// demo
enqueue('A');
enqueue('B');
console.log(dequeue());  // ['A']
console.log(queue);  // ['B']

File: lab/python/tecnicas/fila_enqueue_dequeue.py
fila = []

while True:
    execucao = input("[enqueue | dequeue | exit] ").lower().strip()
    if execucao == "exit":
        print("FIM")
        break
    elif execucao == "enqueue":
        valor = input("Valor a entrar na fila: ")
        fila.append(valor)
    elif execucao == "dequeue":
        if fila:
            fila.pop(0)
        else:
            print("Lista vazia")
    else:
        print("tipo inválido, digite apenas as opções citadas")


print(f"Fila final: {fila}")
